pkg/oam/util/helper.go

```diff
@@ -261,7 +261,8 @@ func fetchChildResources(ctx context.Context, mLog logr.Logger, r client.Reader,
 // PatchCondition condition for a conditioned object
 func PatchCondition(ctx context.Context, r client.StatusClient, workload ConditionedObject,
 	condition ...cpv1alpha1.Condition) error {
-	workloadPatch := client.MergeFrom(workload)
+	patch := workload.DeepCopyObject().(client.Object)
```

Comment:
This needs a comment about why it is required.

Reply:
Added.